### DISCUSSION TITLE
How do I pass the trace ID collected from OTLP to Loki?

### DISCUSSION THREAD

**Discussion by trajano** [open-telemetry/opentelemetry-collector-contrib#29079] [Q&A]
After a bit of discovery on [how to get OTLP logs to be sent to OTEL Collector when using Java Instrumentation](https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9819) I got this message in the debug logs

```
2023-11-09T01:13:33.416Z        info    LogsExporter    {"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": 1, "log records": 1}
2023-11-09T01:13:33.416Z        info    ResourceLog #0
Resource SchemaURL: https://opentelemetry.io/schemas/1.21.0
Resource attributes:
     -> container.id: Str(4e16b1c30e6d5e212d94d518539c579efd34196234588e0ca5e217d49584a416)
     -> host.arch: Str(aarch64)
     -> host.name: Str(4e16b1c30e6d)
     -> os.description: Str(Linux 6.1.21-v8+)
     -> os.type: Str(linux)
     -> process.command_args: Slice(["/usr/lib/jvm/java-17-amazon-corretto/bin/java","-XX:+ExitOnOutOfMemoryError","-Xshare:off","org.springframework.boot.loader.JarLauncher"])
     -> process.executable.path: Str(/usr/lib/jvm/java-17-amazon-corretto/bin/java)
     -> process.pid: Int(1)
     -> process.runtime.description: Str(Amazon.com Inc. OpenJDK 64-Bit Server VM 17.0.9+8-LTS)
     -> process.runtime.name: Str(OpenJDK Runtime Environment)
     -> process.runtime.version: Str(17.0.9+8-LTS)
     -> service.name: Str(auth)
     -> telemetry.auto.version: Str(1.31.0)
     -> telemetry.sdk.language: Str(java)
     -> telemetry.sdk.name: Str(opentelemetry)
     -> telemetry.sdk.version: Str(1.31.0)
ScopeLogs #0
ScopeLogs SchemaURL: 
InstrumentationScope net.trajano.simpleauth.RootController 
LogRecord #0
ObservedTimestamp: 2023-11-09 01:13:32.957827282 +0000 UTC
Timestamp: 2023-11-09 01:13:32.957 +0000 UTC
SeverityText: INFO
SeverityNumber: Info(9)
Body: Str(whomami 2 requested)
Attributes:
     -> loki.format: Str(raw)
     -> loki.attribute.labels: Str(exporter,detected,level,service_name,stream)
Trace ID: b164b1107bfbe43ff5f1b1deab5cd8f4
Span ID: f94db84bc90c03f2
Flags: 1
        {"kind": "exporter", "data_type": "logs", "name": "debug"}
```

Now my question is the **Trace ID** and **Span ID** too.  How do I connect that to Loki so I can query it?

I think it should be done automatically... https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/7569

**Comment by trajano** [comment]
I looked at the logs no trace ID

![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/110627/63d6afe2-5a11-49f3-a041-f84f8d5ef4c6)

but in all honesty if they made the trace ID get sent as a label that is actually bad as there is a high cardinality.

**Comment by trajano** [comment]
Seems like I have to transmit it as JSON to get the traceid as it is neither a resource or attribute but regardless without the support of structured metadata and having it transmitted as a label it will be bad on Loki
![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/110627/7cf2334c-c9c8-49d6-b940-85fb9d371ed7)

**Comment by riteshsonawane1372** [comment]
@trajano found any solution I’m stuck with same issue

**Reply by trajano** [reply]
Sorry. I stopped working on this.
